pkg/utils/utils.go

```diff
@@ -599,3 +599,35 @@ func AbsTarPath(tarPath string) (string, error) {
 
 	return absTarPath, nil
 }
+
+// FindFolder will find the folder 'targetFolder' under the 'root' directory recursively. 
+func FindFolder(root, targetFolder string) (string, error) {
+	var result string
+
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() && strings.EqualFold(info.Name(), targetFolder) {
```

Comment:
Hi @officialasishkumar 😄 

This function needs to look for the package name, not the name of the file directory, here you need to compare the two points, 1. Check whether `kcl.mod` exists in the file path. 2. Whether the package name in `kcl.mod` is the package name you are looking for.

```
if <there is a kcl.mod under path> && <the package name in kcl.mod matched> {
            return <the kcl.mod dir>
}
```

Reply:
> Hi @officialasishkumar 😄 
> 
> This function needs to look for the package name, not the name of the file directory, here you need to compare the two points, 1. Check whether `kcl.mod` exists in the file path. 2. Whether the package name in `kcl.mod` is the package name you are looking for.
> 
> ```
> if <there is a kcl.mod under path> && <the package name in kcl.mod matched> {
>             return <the kcl.mod dir>
> }
> ```
> 

wdym by package name

Reply:
I have updated the file accordingly. Please have a look @Peefy @zong-zhe!